src/async_fuse/memfs/cache/memory_cache.rs

```diff
@@ -2,27 +2,285 @@
 
 use async_trait::async_trait;
 use clippy_utilities::OverflowArithmetic;
+use hashlink::LinkedHashSet;
 use lockfree_cuckoohash::{pin, LockFreeCuckooHash as HashMap};
-use std::collections::HashMap as StdHashMap;
-use std::sync::Arc;
-use tokio::sync::RwLock;
+use std::collections::{BTreeSet, HashMap as StdHashMap};
+use std::mem;
+use std::sync::{Arc, Weak};
+use std::time::Duration;
+use tokio::select;
+use tokio::sync::{mpsc, oneshot, RwLock};
+use tracing::warn;
 
 use super::policy::EvictPolicy;
 use super::{Block, BlockCoordinate, BlockId, Storage};
 use crate::async_fuse::fuse::protocol::INum;
 
-/// The file-level cache.
-type FileCache = Arc<RwLock<StdHashMap<BlockId, Block>>>;
-
 /// Merge the content from `src` to `dst`. This will set `dst` to be dirty.
 fn merge_two_blocks(src: &Block, dst: &mut Block) {
-    dst.set_dirty();
+    dst.set_dirty(true);
     dst.update(src);
 }
 
+/// A command sent from `MemoryCache` to the write back task.
+enum Command {
+    /// Store a block to the backend later.
+    Store {
+        /// The coordinate of the block
+        coord: BlockCoordinate,
+        /// The block
+        block: Block,
+        /// A sender to notify the `MemoryCache` that the block has been flushed to the backend
+        sender: oneshot::Sender<()>,
+    },
+    /// Flush the file specified immediately.
+    Flush(INum),
+    /// Flush all the files
+    FlushAll(oneshot::Sender<()>),
+}
+
+/// A function to run a write back task.
+///
+/// This task will smoothly write blocks back to the backend
+/// when the cache runs in write-back policy.
+///
+/// And this task will evict blocks to the backend if the number of blocks hits the soft limit,
+/// whether the cache runs in write-back policy or not.
+async fn write_back_task<P, S>(
```

Comment:
This function is too long. It should be broken down into three sub-functions:

1. The update of the period.
2. The handling of commands, each type of command should have its own handling function.
3. Is the handling of an expired interval also considered a type of command?

Reply:
3. No, it's not a type of command. A command is just a message sent from the cache. But put its handling into a sub-function is a good idea.